server/src/termdb.js

```diff
@@ -494,15 +493,23 @@ async function get_AllSamplesByName(q, req, res, ds) {
 		const lst = await ds.queries.singleCell.samples.get()
 		const result = {}
 		for (const s of lst) {
-			// {name}
 			result[s.sample] = ds.cohort.termdb.q.sampleName2id(s.sample)
 		}
 		res.send(result)
 		return
-	}
+	} else {
+		let sampleName2Id = ds.sampleName2Id
+		if (q.filter) {
+			q.ds = ds
```

Comment:
no need to attach ds to q?

Reply:
it is actually needed, if you comment out that line it fails. It comes from the line 76, 
`// needed by some helper functions
q.ds = ds`